# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [2]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [3]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [5]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [6]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [7]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [8]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [9]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [10]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [11]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [12]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [13]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [14]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [15]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [16]:
data.duplicated().sum()

71

In [17]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [1]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [19]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [20]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [21]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [22]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

##### Посчитаем общее количество всех кто брал кредит и количество тех кто не вернул, с учетом количества детей. 
##### Для этого построим сводную таблицу по столбцам 'children' (количество детей) и 'debt' (задолженность по возврату кредита). Посчитаем общее количество методом группировки. 
##### Найдем долю не вернувших кредит в каждой категории разделив количество не вернувших на общее количество.

In [23]:
data_pivot_child = data.pivot_table(index=['children'], values='debt', aggfunc = 'sum')
child_debt = data.groupby('children')['debt'].count()

print(data_pivot_child)
print(child_debt)

print(data_pivot_child['debt']/child_debt*100)

          debt
children      
0         1063
1          444
2          194
3           27
4            4
5            0
children
0    14091
1     4808
2     2052
3      330
4       41
5        9
Name: debt, dtype: int64
children
0    7.543822
1    9.234609
2    9.454191
3    8.181818
4    9.756098
5    0.000000
Name: debt, dtype: float64


**Вывод:**  
Больше всего кредит берут люди без детей (14 000 против 4800).  
И в тоже время у людей без детей самый низкий процент не возврата по кредиту (7,54 %).  
С количеством детей количество кредита уменьшается. Меньше всего кредитов берут люди с 3-4 детьми и больше.  
С появлением хотя бы одного ребенка процент не возврата по кредиту увеличивается (9,23% против 7,54%).  
А вот дальше появление в семье второго, третьего ребенка и т.д практически не влияют на долю не возврата.

##### Построим сводную таблицу по столбцам 'children' (количество детей) и 'debt' (задолженность по возврату кредита). В этой таблице выведем общее количество кредитов, количество просрочки и долю не возврата кредита в срок используя функцию aggfunc=

In [74]:
data_child = data.pivot_table(index=['children'], values='debt', aggfunc= {'debt':['sum', 'count', 'mean']})

data_child=data_child.rename(columns={'count':'количество кредитов', 'mean':'доля просрочки', 'sum':'количество просрочки'})

data_child['доля просрочки'] = data_child['доля просрочки']*100

print(data_child)

          количество кредитов  доля просрочки  количество просрочки
children                                                           
0                     14091.0        7.543822                1063.0
1                      4808.0        9.234609                 444.0
2                      2052.0        9.454191                 194.0
3                       330.0        8.181818                  27.0
4                        41.0        9.756098                   4.0
5                         9.0        0.000000                   0.0


#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

##### Как и в предыдущем выводе построим сводную таблицу и воспользуемся группировкой.
##### Построим сводную таблицу по столбцам 'family_status' (семейное положение) и 'debt' (задолженность по возврату кредита) для подсчета количества не возврата кредита в зависимости от семейного положения.
##### Сгруппируем по 'family_status' и посчитаем общее количество всех кто брал кредит в зависимости от семейного положения.
##### Найдем долю невозврата кредита разделив значения из сводной таблицы на общее количество

In [69]:
data_pivot_fam = data.pivot_table(index=['family_status'], values='debt', aggfunc = 'sum')
fam_debt = data.groupby('family_status')['debt'].count()

print(data_pivot_fam)
print()
print(fam_debt)
print()
print(data_pivot_fam['debt']/fam_debt*100)

                       debt
family_status              
Не женат / не замужем   273
в разводе                84
вдовец / вдова           63
гражданский брак        385
женат / замужем         927

family_status
Не женат / не замужем     2796
в разводе                 1189
вдовец / вдова             951
гражданский брак          4134
женат / замужем          12261
Name: debt, dtype: int64

family_status
Не женат / не замужем    9.763948
в разводе                7.064760
вдовец / вдова           6.624606
гражданский брак         9.313014
женат / замужем          7.560558
Name: debt, dtype: float64


##### Построим сводную таблицу по столбцам 'family_status' (семейное положение) и 'debt' (задолженность по возврату кредита). Также как и в предыдущем примере выведем общее количество кредитов, количество просрочки и долю не возврата кредита в срок используя функцию aggfunc=

In [76]:
data_fam = data.pivot_table(index=['family_status'], values='debt', aggfunc= {'debt':['sum', 'count', 'mean']})

data_fam=data_fam.rename(columns={'count':'количество кредитов', 'mean':'доля просрочки', 'sum':'количество просрочки'})

data_fam['доля просрочки'] = data_fam['доля просрочки']*100

print(data_fam)

                       количество кредитов  доля просрочки  \
family_status                                                
Не женат / не замужем               2796.0        9.763948   
в разводе                           1189.0        7.064760   
вдовец / вдова                       951.0        6.624606   
гражданский брак                    4134.0        9.313014   
женат / замужем                    12261.0        7.560558   

                       количество просрочки  
family_status                                
Не женат / не замужем                 273.0  
в разводе                              84.0  
вдовец / вдова                         63.0  
гражданский брак                      385.0  
женат / замужем                       927.0  


**Вывод:**  
Больше всех кредиты берут женатые/замужние (12261). Меньше всех кредит берут вдовцы (951).  
Самый низкий процент не возврата у вдовцов (6,62%)  
У не женатых/не замужних и людей в гражданском браке доля не возврата примерно одинаковая и самая высокая в выборке (9,3% - 9,7%)  
Также и людей в разводе и женатых доля не возврата примерно одинаковая, но ниже чем у не женатых и в гражданском браке (7,5% - 7,04%)  

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

##### По аналогии с прошлыми выводами построим свобдную таблицу и сгруппируем по 'total_income_category'
##### Категории доходов:  
 - 0–30000 — 'E';
- 30001–50000 — 'D';
- 50001–200000 — 'C';
- 200001–1000000 — 'B';
- 1000001 и выше — 'A'.

In [70]:
data_pivot_total = data.pivot_table(index=['total_income_category'], values='debt', aggfunc = 'sum')
total_debt = data.groupby('total_income_category')['debt'].count()

print(data_pivot_total)
print()
print(total_debt)
print()
print(data_pivot_total['debt']/total_debt*100)

                       debt
total_income_category      
A                         2
B                       354
C                      1353
D                        21
E                         2

total_income_category
A       25
B     5014
C    15921
D      349
E       22
Name: debt, dtype: int64

total_income_category
A    8.000000
B    7.060231
C    8.498210
D    6.017192
E    9.090909
Name: debt, dtype: float64


In [77]:
data_total = data.pivot_table(index=['total_income_category'], values='debt', aggfunc= {'debt':['sum', 'count', 'mean']})

data_total=data_total.rename(columns={'count':'количество кредитов', 'mean':'доля просрочки', 'sum':'количество просрочки'})

data_total['доля просрочки'] = data_total['доля просрочки']*100

print(data_total)

                       количество кредитов  доля просрочки  \
total_income_category                                        
A                                     25.0        8.000000   
B                                   5014.0        7.060231   
C                                  15921.0        8.498210   
D                                    349.0        6.017192   
E                                     22.0        9.090909   

                       количество просрочки  
total_income_category                        
A                                       2.0  
B                                     354.0  
C                                    1353.0  
D                                      21.0  
E                                       2.0  


**Вывод:**  
Больше всего людей берут кредит с зарплатой от 50 001 до 200 000 руб.  
Меньше всего людей берут кредит с зарплатой свыше миллиона рублей и ниже 30 000 рублей.  
Притом самый высокий процент не возврата у людей с зарплатой меньше 30 000 руб (9,09%).  
Самый низкий процент не вовзврата в категории зарплат от 30 001 до 50 000 руб (6,01%).  
Не возврат кредита у людей с зарплатами выше миллиона и от 50 001 до 200 000 примерно одинаковый (8% - 8,4%).  
Люди с зарплатами от 200 001 до миллиона не очень активно берут кредит и в тоже время стараются вернуть его в срок. процент не возврата средний (7,06%)

#### Вывод  
Категории В и С значительно отличаются по количеству кредитов среди отстальных.  
Поэтому для анализа будем использовать только эти две категории, т.е людей с зарплатой от 50 000 до 200 000 и от 200 001 до 1 000 000.  
Остальные категории не окажут существенного влияния на показатели.  
Люди с зарплатой от 50 000 до 200 000 берут почти в 3 раза больше берут кредит, чем с зарплатой от 200 001 до 1 000 000.  
И в тоже время, доля просрочки по кредиту у людей с зарплатой от 50 000 до 200 000 выше.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

##### Проведем тоже исследование для целей кредита. На этот раз возьмем столбец purpose_category (категории целей)

In [71]:
data_pivot_purpose = data.pivot_table(index=['purpose_category'], values='debt', aggfunc = 'sum')
purpose_debt = data.groupby('purpose_category')['debt'].count()

print(data_pivot_purpose)
print()
print(purpose_debt)
print()
print(data_pivot_purpose['debt']/purpose_debt*100)

                          debt
purpose_category              
операции с автомобилем     400
операции с недвижимостью   780
получение образования      369
проведение свадьбы         183

purpose_category
операции с автомобилем       4279
операции с недвижимостью    10751
получение образования        3988
проведение свадьбы           2313
Name: debt, dtype: int64

purpose_category
операции с автомобилем      9.347978
операции с недвижимостью    7.255139
получение образования       9.252758
проведение свадьбы          7.911803
Name: debt, dtype: float64


In [80]:
data_purpose = data.pivot_table(index=['purpose_category'], values='debt', aggfunc= {'debt':['sum', 'count', 'mean']})

data_purpose=data_purpose.rename(columns={'count':'количество кредитов', 'mean':'доля просрочки', 'sum':'количество просрочки'})

data_purpose['доля просрочки'] = data_purpose['доля просрочки']*100

print(data_purpose)

                          количество кредитов  доля просрочки  \
purpose_category                                                
операции с автомобилем                 4279.0        9.347978   
операции с недвижимостью              10751.0        7.255139   
получение образования                  3988.0        9.252758   
проведение свадьбы                     2313.0        7.911803   

                          количество просрочки  
purpose_category                                
операции с автомобилем                   400.0  
операции с недвижимостью                 780.0  
получение образования                    369.0  
проведение свадьбы                       183.0  


**Вывод:**  
Чаще всего люди берут кредит на операции с недвижимостью.  
При этом процент не возврата по этим целям самый низкий (7,2%).  
Меньше всего люди берут кредит на проведение свадьбы и процент не возврата почти такой же низкий как с недвижимостью (7,2%).  
Самая большая просрочка по кредиту у людей бравших кредит на образование и операции с автомобилем (9,2% - 9,3%).

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

**Ответ:** Пропуски в исходных данных были в информации об общем трудовом стаже и ежемесячном доходе. Скорее всего потенциальные заемщики специально не указывали эту информацию, чтобы повысить вероятность одобрения кредита.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

**Ответ:** При учете того, что в данных были зарплаты свыше миллиона и большой трудовой стаж, среднее арифмитическое по этим значениям могло сильно исказить данные. Поэтому медианные значение были наиболее подходящим вариантом.

### Шаг 4: общий вывод.

 ### Общий вывод
 В полученных данных были обнаружены не заполненные значения в общем трудовом стаже и ежемесячном доходе. Есть вероятность, что некоторые заемщики специально не указывали свой доход и стаж работы для повышения вероятности одобрения кредита. 
Пропущенные данные были заполнены медианными значения, так как в трудовом стаже и зарплате присутствуют сильно выделяющиеся значения и среднее арифмитическое по этим значениям могло сильно исказить данные. В то время как медиана устойчива к выделющимся значениям и выбросам.
    
Также в данных встречались отрицательные значения трудового стажа, которые могли появиться по ошибке ввода или технической ошибке. Такие данные были приведены к положительным значениям.
    
В информации о детях встречались аномальные значения, отрицательное значение или слишком большое (20). Также в данных присутствовали повторяющиеся значения. Аномальные и повторяющиеся значения были удалены для проведения корректного анализа.  
    
Для анализа данных информацию о зарплате привели к категориям где зарплаты:  
- 0–30000 — 'E';
- 30001–50000 — 'D';
- 50001–200000 — 'C';
- 200001–1000000 — 'B';
- 1000001 и выше — 'A'.
    
    
Цели кредита тоже были приведены к категориям: 'операции с автомобилем', 'операции с недвижимостью', 'проведение свадьбы',
'получение образования'.
    
### Вывод по исследованию
    
Семейное положение влияет на факт погашения кредита в срок.  
    
Как показали данные, самый низкий процент не возврата у вдовцов (6,62%).  
У не женатых/не замужних и людей в гражданском браке доля не возврата примерно одинаковая и самая высокая в выборке (9,3% - 9,7%).  
У людей в разводе и женатых доля не возврата примерно одинаковая, но ниже чем у не женатых и в гражданском браке (7,5% - 7,04%).  
    
Количество детей тоже влияет на факт погашения кредита в срок.  
У людей без детей самый низкий процент не возврата по кредиту (7,54 %).  
У людей с одним и двумя детьми процент не возврата по кредиту увеличивается и составляет 9,2% - 9,4%.  
Доля людей берущих кредит с тремя и более детьми слишком мала, поэтому этими категориями можно пренебречь.  
    
Также, в ходе исследования было определено, что люди с зарплатой от 50 000 до 200 000 рублей берут кредит почти в 3 раза чаще, чем с зарплатой от 200 001 до 1 000 000 рублей.  
При этом, доля просрочки по кредиту у людей с зарплатой от 50 000 до 200 000 выше и составляет 8,5%.  
Доля просрочки у людей с зарплатой от 200 001 до 1 000 000 рублей составляет 7,06%.  
Доля зарплат в категориях С и В составляет порядка 98% от всех данных, поэтому остальными категориями зарплат можно пренебречь.
    
Цели кредита также показали влияние на факт погашения кредита в срок.  
Так, самый низкий процент не возврата кредита в срок у людей берущих кредит на операции с недвижимостью и свадьбы 7,2% - 7,9%.  
У людей берущих кредит на автомобили и образование просрочка по кредиту составляет 9,2% - 9,3%.